sycl/doc/EnvironmentVariables.md

```diff
@@ -149,6 +149,7 @@ variables in production code.</span>
 | `SYCL_PI_LEVEL_ZERO_USE_COPY_ENGINE` | Any(\*) | This environment variable enables users to control use of copy engines for copy operations. If the value is an integer, it will allow the use of copy engines, if available in the device, in Level Zero plugin to transfer SYCL buffer or image data between the host and/or device(s) and to fill SYCL buffer or image data in device or shared memory. The value of this environment variable can also be a pair of the form "lower_index:upper_index" where the indices point to copy engines in a list of all available copy engines. The default is 1. |
 | `SYCL_PI_LEVEL_ZERO_USE_COPY_ENGINE_FOR_D2D_COPY` (experimental) | Integer | Allows the use of copy engine, if available in the device, in Level Zero plugin for device to device copy operations. The default is 0. This option is experimental and will be removed once heuristics are added to make a decision about use of copy engine for device to device copy operations. |
 | `SYCL_PI_LEVEL_ZERO_DEVICE_SCOPE_EVENTS` | Any(\*) | Enable support of device-scope events whose state is not visible to the host. If enabled mode is SYCL_PI_LEVEL_ZERO_DEVICE_SCOPE_EVENTS=1 the Level Zero plugin would create all events having device-scope only and create proxy host-visible events for them when their status is needed (wait/query) on the host. If enabled mode is SYCL_PI_LEVEL_ZERO_DEVICE_SCOPE_EVENTS=2 the Level Zero plugin would create all events having device-scope and add proxy host-visible event at the end of each command-list submission. The default is 0, meaning all events are host-visible. |
+| `SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS` | Integer | When set to a positive value enables use of Level Zero immediate commandlists. If enabled, it is necessary to also set SYCL_PI_LEVEL_ZERO_DEVICE_SCOPE_EVENTS=1. Default is 0. |
```

Comment:
Please clarify in the description that in this mode commands are immediately submitted for execution, without any batching. Also SYCL_PI_LEVEL_ZERO_DEVICE_SCOPE_EVENTS=0 should be working as well. BTW, I noticed that the default of SYCL_PI_LEVEL_ZERO_DEVICE_SCOPE_EVENTS is still saying "0", which is not true, it's "2". Please fix here if you don't mind.

Reply:
OK.

Reply:
Hi. I think it might be a good idea to change the default to 2 in a separate patch. That way, we will be able to identify any performance improvement/regression due to that change in an easier way. Thanks

Reply:
I asked to do this change here. The default is already "2", it's just the documentation that wasn't updated.

Reply:
Got it...thanks